src/main/java/org/snomed/snowstorm/core/data/domain/Relationship.java

```diff
@@ -187,30 +187,28 @@ public boolean isGrouped () {
 	}
 
 	public ConcreteValue getConcreteValue() {
-		if (this.value == null) {
-			return null;
-		}
-
-		this.concreteValue = new ConcreteValue(this.value);
 		return this.concreteValue;
 	}
 
 	public void setConcreteValue(ConcreteValue concreteValue) {
 		this.concreteValue = concreteValue;
 	}
 
-	public void setConcreteValue(String value) {
-		if (value != null) {
-			this.concreteValue = new ConcreteValue(value);
+	public void setConcreteValueFromExternal(org.snomed.otf.owltoolkit.domain.Relationship.ConcreteValue concreteValue) {
+		if (concreteValue != null) {
+			this.concreteValue = ConcreteValue.from(concreteValue);
+			this.value = concreteValue.getRF2Value();
+			this.destinationId = null;
+			this.target = null;
 		}
-
-		this.destinationId = null;
-		this.value = value;
-		this.target = null;
 	}
 
 	public boolean isConcrete() {
```

Comment:
The isConcrete method was situation dependent, so I changed the implementation to try and cater for the situations.

Reply:
Like @CoderMChu suggested, the destinationId is now set to null in the constructor; the implementation of isConcrete is now much simpler.